spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1038,6 +1038,46 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+  test("hex") {
+    val str_table = "string_hex_table"
+    withTable(str_table) {
```

Comment:
Let's add more types and values, something like `makeRawTimeParquetFile`

Reply:
Thanks, will have a look at `makeRawTimeParquetFile` and follow up.

Reply:
probably `makeParquetFileAllTypes` is a better example

Reply:
Thanks for the feedback. I update the tests in https://github.com/apache/datafusion-comet/pull/449/commits/2bfcf25a6705ec47d857670c65dbba979ca10e23 to use `makeParquetFileAllTypes`. And with a couple of changes its tests passes w/ the ints, uint32 and uint64[1], and the _8 and _14 which I think are string/binary types.

When the #451 goes in, I'll look at updating the tests to add it too.

[1]: I'm a bit confused because it looks like every int crossing into rust does so as an i64, which means rather than properly return say `FF` for u8, I'm returning `FFFFFFFFFFFFFFFF`. I need to look into this further.

Reply:
@advancedxy got a good point. Spark may be casting ints to `i64`. Then we only need to handle `i64` in Rust as long as it passes the tests.

Reply:
Yeah, I think that's the case. I'll pare down the rust code.